File: 백준/Gold/17251. 힘 겨루기/힘 겨루기.py
import sys
read = sys.stdin.readline

n = int(read())
arr = list(map(int, read().split()))
left_max = [0] * n
right_max = [0] * n
left_max[0] = arr[0]
right_max[n-1] = arr[n-1]

for i in range(1, n):
    left_max[i] = max(left_max[i-1], arr[i])
    right_max[n-1-i] = max(right_max[n-i], arr[n-1-i])

red_cnt = 0
blue_cnt = 0
for i in range(n-1):
    if left_max[i] > right_max[i+1]:
        red_cnt += 1
    elif left_max[i] < right_max[i+1]:
        blue_cnt += 1

if red_cnt > blue_cnt:
    print('R')
elif red_cnt < blue_cnt:
    print('B')
else:
    print('X')

File: 백준/Gold/17251. 힘 겨루기/힘 겨루기.js
const input = require('fs').readFileSync(process.platform === "linux" ? "/dev/stdin" : "./sample.txt").toString().trim().split("\n");

const n = Number(input[0].split(' '));
const arr = input[1].split(' ').map(Number);

const leftMax = [...Array(n)].map((v) => 0);
let redWins = 0;

const rightMax = [...Array(n)].map((v) => 0);
let blueWins = 0;

leftMax[0] = arr[0];
rightMax[n - 1] = arr[n - 1];

for (let i = 1; i < n; i++) {
  leftMax[i] = Math.max(leftMax[i - 1], arr[i]);
  rightMax[n - 1 - i] = Math.max(rightMax[n - i], arr[n - 1 - i]);
}

for (let i = 0; i < n - 1; i++) {
  if (leftMax[i] > rightMax[i + 1]) { redWins += 1; }
  if (leftMax[i] < rightMax[i + 1]) { blueWins += 1; }
}

if (redWins > blueWins) { console.log('R'); }
else if (redWins < blueWins) { console.log('B'); }
else { console.log('X'); }